### PR TITLE
Fix: cannot create env0_user_team_assignment: validation error

### LINKED ISSUE #681: Cannot create env0_user_team_assignment: validation error
My userId contains an `_` in it and for that reason I'm unable to create an env0_user_team_assignment resource.
The following function fails because two underlines are present in the string.
https://github.com/env0/terraform-provider-env0/blob/9db74b01fa8a46142d8768195a2303fa64c4144c/env0/resource_user_team_assignment.go#L36-L45

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
fixes #681 

### Solution

Some user ids have an "_". Original solution assumed it's not possible.

1. created a custom split called lastSplit: split by the last underscore.
2. use lastSplit instead of using strings.Split().
3. Updated the acceptance tests to cover this use-case.
4. Added unit tests for lastSplit.

### CODE REVIEW

File: env0/resource_user_team_assignment.go [hunk 1]
```diff
@@ -34,7 +33,7 @@ func (a *UserTeamAssignment) GetId() string {
 }
 
 func GetUserTeamAssignmentFromId(id string) (*UserTeamAssignment, error) {
-	splitUserTeam := strings.Split(id, "_")
+	splitUserTeam := lastSplit(id, "_")
```
Comment (TomerHeber) [inline env0/resource_user_team_assignment.go:36]: At the moment this is the only code where import is used with a combination of "user".
If more combinations are added need to be sure lastSplit is used.

File: env0/resource_user_team_assignment.go [hunk 2]
```diff
@@ -34,7 +33,8 @@ func (a *UserTeamAssignment) GetId() string {
 }
 
 func GetUserTeamAssignmentFromId(id string) (*UserTeamAssignment, error) {
-	splitUserTeam := strings.Split(id, "_")
+	// lastSplit is used to avoid issues where the user_id has underscores in it.
+	splitUserTeam := lastSplit(id, "_")
 	if len(splitUserTeam) != 2 {
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/resource_user_team_assignment.go:38]: in golang `len(nil)` returns `0`.

File: env0/utils.go [hunk 3]
```diff
@@ -489,3 +489,12 @@ func ttlToDuration(ttl *string) (time.Duration, error) {
 
 	return time.ParseDuration(fmt.Sprintf("%dh", hours))
 }
+
+func lastSplit(s string, sep string) []string {
+	lastIndex := strings.LastIndex(s, sep)
+	if lastIndex == -1 {
+		return nil
```
Comment (sagilaufer1992) [inline env0/utils.go:496]: why `nil`? the original split would return the original string, no? that might confuse when using that method.
in the case of this fix, the outcome will remain that the length of the desired array won't be 2, so IMO it's better to stick to known outcomes

File: env0/resource_user_team_assignment.go (hunk 1 above)
Comment (sagilaufer1992) [reply env0/resource_user_team_assignment.go:37]: just to make sure - team id is some kind of guid, so it won't have `_` in it, so we won't have to consider this case as well?

**Review by sagilaufer1992 — COMMENTED** [review]
looks good overall, left some comments

File: env0/utils.go (hunk 3 above)
Comment (TomerHeber) [reply env0/utils.go:496]: fair point. I'll change that.

File: env0/resource_user_team_assignment_test.go [hunk 4]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 func TestUnitUserTeamAssignmentResource(t *testing.T) {
-	userId := "uid"
+	userId := "ui_d"
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/resource_user_team_assignment_test.go:16]: I can duplicate the test with a different value.

File: env0/resource_user_team_assignment.go (hunk 1 above)
Comment (TomerHeber) [reply env0/resource_user_team_assignment.go:37]: AFAIK yes - a uuid.
userid is not a uuid.

File: env0/resource_user_team_assignment_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/resource_user_team_assignment_test.go]: both use-cases are now covered (added another test).

File: env0/utils.go (hunk 3 above)
Comment (TomerHeber) [reply env0/utils.go]: fixed.

**Review by sagilaufer1992 — APPROVED** [review]
LGTM, my comments were addressed

**Comment by sagilaufer1992** [comment]
please try and comment your commit hash that resolves a comment, it makes reviewing easier 😃

**Pull request merged**